iroh/src/sync_engine.rs

```diff
@@ -243,11 +243,15 @@ impl SyncEngine {
     }
 }
 
-pub(crate) fn entry_to_content_status(entry: EntryStatus) -> ContentStatus {
+pub(crate) fn entry_to_content_status(entry: io::Result<EntryStatus>) -> ContentStatus {
     match entry {
-        EntryStatus::Complete => ContentStatus::Complete,
-        EntryStatus::Partial => ContentStatus::Incomplete,
-        EntryStatus::NotFound => ContentStatus::Missing,
+        Ok(EntryStatus::Complete) => ContentStatus::Complete,
+        Ok(EntryStatus::Partial) => ContentStatus::Incomplete,
+        Ok(EntryStatus::NotFound) => ContentStatus::Missing,
+        Err(cause) => {
+            tracing::warn!("Error while checking entry status: {cause:?}");
+            ContentStatus::Missing
```

Comment:
one other thought, maybe it might make sense to introduce `ContentStatus::Unknown` for cases like this at some point

Reply:
right, but if the report is `missing` you won't know to do that, right?

Reply:
Yeah, the proper way to deal with this is to make this a Result<ContentStatus> or have some other way to signal to the sync engine that "something's wrong, please stop what you're doing".

But I would prefer to do this in a subsequent PR.